src/sensesp_ble_gateway/ble_signalk_gateway.cpp

```
@@ -229,10 +229,19 @@ void BLESignalKGateway::on_advertisement() {
     adv_dropped_count_.fetch_add(1, std::memory_order_relaxed);
     return;
   }
-  if (pending_ads_.size() >= config_.max_pending_ads) {
+  // Bound by the capacity actually reserved, not only by the configured
+  // maximum. Between a drain and restore_pending_capacity() the array is
+  // reserved for the window rather than the whole buffer, and pushing past
+  // that would reallocate here — on the Bluetooth host's callback, in a
+  // build without exceptions, where a failed allocation aborts the device.
+  // Trimming instead keeps the push non-allocating on every path.
+  const size_t capacity = pending_ads_.capacity() < config_.max_pending_ads
```

Clang (14.0.6)

[warning] variable 'capacity' is not initialized

(cppcoreguidelines-init-variables)